lib/hanami/configuration.rb

```diff
@@ -50,10 +51,15 @@ def logger(options = nil)
       if options.nil?
         settings.fetch(:logger)
       else
-        settings.put_if_absent(:logger, options)
+        settings[:logger] = options
       end
     end
 
+    def environment(name)
+      # settings[:environments][name.to_sym] << blk
+      yield if ENV['HANAMI_ENV'] == name.to_s
```

Comment:
I don't think so. The `Hanami::Environment` is created after all the components are resolved and that happens prior to that.